pkg/gds/models/v1/models.go

```diff
@@ -172,6 +172,28 @@ func AppendAdminEmailLog(vasp *pb.VASP, reason string, subject string) (err erro
 	return nil
 }
 
+func GetSentAdminEmailCount(vasp *pb.VASP, reason string, timeWindowDays int) (sent int, err error) {
+	var adminEmailLog []*EmailLogEntry
+	if adminEmailLog, err = GetAdminEmailLog(vasp); err != nil {
+		return 0, err
+	}
+
+	for _, value := range adminEmailLog {
+		var timestamp time.Time
+		if timestamp, err = time.Parse(time.RFC3339, value.Timestamp); err != nil {
+			return 0, fmt.Errorf("error parsing timestamp: %v", err)
+		}
+
+		matchedReason := reason == value.Reason
+		withinTimeWindow := timestamp.After(time.Now().AddDate(0, 0, -timeWindowDays))
+
+		if matchedReason && withinTimeWindow {
+			sent++
+		}
+	}
+	return sent, nil
+}
```

Comment:
There's some duplicate code between this method and the one in `contacts.go`, however it's fine if we have to open a separate story to do the cleanup.